### PR TITLE
wip: feat(linter): add eslint/no-explicity-any

### PR DESCRIPTION
**Pull request by manuschillerdev**
This implements the `no-explicit-any` rule for the typescript linter.

* Documentation: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-explicit-any.md
* Tests: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/tests/rules/no-explicit-any.test.ts 
* Rules: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/no-explicit-any.ts

- [x] implement basic checks
- [x] add basic test cases
- [x] add configuration options
- [ ] branch linting via configuration options
- [x] use configuration options in testcases
- [ ] check against contribution guidelines of the project :)

### CODE REVIEW

File: crates/oxc_linter/src/rules/typescript/no_explicit_any.rs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        Self { ignore_rest_args }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        "const number: number = 1;",
```
Comment (manuschillerdev) [inline crates/oxc_linter/src/rules/typescript/no_explicit_any.rs:73]: todo: use tuples with configuration options for test cases. See: https://github.com/Boshen/oxc/blob/main/crates/oxc_linter/src/rules/eslint/getter_return.rs#L189

File: crates/oxc_linter/src/rules/typescript/no_explicit_any.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+"typescript-eslint(no-explicit-any): Using any disables many type checking rules and is generally best used only as a last resort or when prototyping code"
```
Comment (manuschillerdev) [inline crates/oxc_linter/src/rules/typescript/no_explicit_any.rs:13]: todo: check which labels/descriptions would be appropriate here

File: crates/oxc_linter/src/rules/typescript/no_explicit_any.rs [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    /// as a type annotation.
+    ///
+    /// ### Example
+    /// ```typescript
+    /// const age: any = 'seventeen';
+    /// const ages: any[] = ['seventeen'];
+    /// const ages: Array<any> = ['seventeen'];
+    /// ```
+    NoExplicitAny,
+    correctness
```
Comment (manuschillerdev) [inline crates/oxc_linter/src/rules/typescript/no_explicit_any.rs:42]: should probably be `restriction`: 

https://github.com/manuschillerdev/oxc/blob/42b97fc270e2e1601d8f1c93379cda712cd94255/crates/oxc_linter/src/rule.rs#L42

File: crates/oxc_linter/src/rules/typescript/no_explicit_any.rs [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+"typescript-eslint(no-explicit-any): Using any disables many type checking rules and is generally best used only as a last resort or when prototyping code"
+)]
+#[diagnostic(severity(warning))]
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/typescript/no_explicit_any.rs:15]: No diagnostic?

**Comment by DonIsaac** [comment]
@manuschillerdev any plans to complete this PR?

**Pull request closed without merging**